apps/ftu/test/marionette/lib/ftu.js

```diff
@@ -78,6 +103,63 @@ Ftu.prototype = {
       throw new Error('Option '+ language +
                       ' could not be found in select wrapper');
     }
+  },
+
+  getLocationHash: function() {
+    var hash = this.client.executeScript(function() {
+      return window.wrappedJSObject.location.hash;
+    });
+    return hash;
+  },
+
+  goNext: function() {
+    var loadingOverlay = this.client.findElement('#loading-overlay');
+    var finishScreen = this.client.findElement(Ftu.Selectors.finishScreen);
+    this.client.waitFor(function() {
+      return !loadingOverlay.displayed();
+    });
```

Comment:
I need to find the source for this, does it return immediately if its already not displayed? Or rely on a transition/animation